Sofa/framework/LinearAlgebra/src/sofa/linearalgebra/CompressedRowSparseMatrix.h

```diff
@@ -19,1827 +19,17 @@
 *                                                                             *
 * Contact information: contact@sofa-framework.org                             *
 ******************************************************************************/
+/******************************************************************************
+* Contributors:
+*   - InSimo
+*******************************************************************************/
 #pragma once
-#include <sofa/linearalgebra/config.h>
 
-#include <sofa/linearalgebra/BaseMatrix.h>
-#include <sofa/linearalgebra/MatrixExpr.h>
-#include <sofa/linearalgebra/matrix_bloc_traits.h>
-#include <sofa/linearalgebra/FullMatrix.h>
-#include <sofa/linearalgebra/FullVector.h>
-#include <sofa/type/vector.h>
-#include <sofa/helper/rmath.h>
+#include <sofa/linearalgebra/CompressedRowSparseMatrixMechanical.h>
 
 namespace sofa::linearalgebra
 {
+    template<typename TBlock, typename TVecBlock = type::vector<TBlock>, typename TVecIndex = type::vector<sofa::Index> >
+    using CompressedRowSparseMatrix = CompressedRowSparseMatrixMechanical<TBlock, CRSMechanicalPolicy>;
```

Comment:
I see that `TVecBlock` and `TVecIndex` are no longer taken into account. It is replaced by the ones in `CRSBlockTraits`. I don't know if it is a problem, but strictly speaking your new `CompressedRowSparseMatrix` is not equivalent to the previous one.
I suggest to add a mechanism to detect if  `TVecBlock` and `TVecIndex` differ from `CRSBlockTraits`.

Reply:
what do you suggest ? 
I was thinking to create a struct inheriting CRSMecha instead of an alias, to add the static assert inside, but this comes with lots of problems as :
 - the new struct will be a new type (problem with operator & stuff)
 - the `final` keyword in CRSMecha which prevents inhering; I guess we could remove it but I dont know the impact on the perf